### PR TITLE
Remove bus from regulated buses list when the generator's diagram is flat

### PR DESCRIPTION
**Pull request by cossartque**

### CODE REVIEW

**Review by lecourtoisflo — APPROVED** [review]

File: sources/Algo/src/Algo.cpp [hunk 1]
```diff
@@ -115,6 +115,7 @@ GeneratorDefinitionAlgorithm::operator()(const NodePtr& node) {
     if ((model == GeneratorDefinition::ModelType::DIAGRAM_PQ_SIGNALN || model == GeneratorDefinition::ModelType::REMOTE_DIAGRAM_PQ_SIGNALN ||
          model == GeneratorDefinition::ModelType::PROP_DIAGRAM_PQ_SIGNALN || model == GeneratorDefinition::ModelType::WITH_IMPEDANCE_DIAGRAM_PQ_SIGNALN) &&
         !isDiagramValid(generator)) {
+      busesWithDynamicModel_.erase(generator.regulatedBusId);
```
Comment (lecourtoisflo) [inline sources/Algo/src/Algo.cpp:118]: in worst case, we have an insert then an erase, which is more time consuming than performing twice the test (model == ...) in switch case

**Review by lecourtoisflo — CHANGES_REQUESTED** [review]

File: sources/Algo/src/Algo.cpp [hunk 2]
```diff
@@ -87,14 +87,23 @@ GeneratorDefinitionAlgorithm::GeneratorDefinitionAlgorithm(Generators& gens, Bus
 void
 GeneratorDefinitionAlgorithm::operator()(const NodePtr& node) {
   auto& node_generators = node->generators;
+
+  auto is_model_diagram = [](GeneratorDefinition::ModelType model, inputs::Generator generator) {
+    return (model == GeneratorDefinition::ModelType::DIAGRAM_PQ_SIGNALN || model == GeneratorDefinition::ModelType::REMOTE_DIAGRAM_PQ_SIGNALN || 
+            model == GeneratorDefinition::ModelType::PROP_DIAGRAM_PQ_SIGNALN || model == GeneratorDefinition::ModelType::WITH_IMPEDANCE_DIAGRAM_PQ_SIGNALN) && 
+            !isDiagramValid(generator);
+ };
```
Comment (lecourtoisflo) [inline sources/Algo/src/Algo.cpp:95]: alinea is incorrect: please apply formatting

**Review by lecourtoisflo — CHANGES_REQUESTED** [review]
Please do not put names in commit message

File: sources/Algo/src/Algo.cpp (hunk 2 above)
Comment (cossartque) [reply sources/Algo/src/Algo.cpp]: Ok thanks

**Review by lecourtoisflo — APPROVED** [review]

**Review by rosiereflo — APPROVED** [review]

**Review by marcochiaramello — APPROVED** [review]

**Pull request merged**